src/main/java/com/artipie/ResponseMetricsSlice.java

```diff
@@ -87,16 +89,41 @@ public Response response(
      * Report response to metrics.
      *
      * @param rqline Request line.
+     * @param rqheaders Request headers.
      * @param rsstatus Response status.
      */
-    private void report(final String rqline, final RsStatus rsstatus) {
+    private void report(
+        final String rqline,
+        final Iterable<Map.Entry<String, String>> rqheaders,
+        final RsStatus rsstatus
+    ) {
         if (rsstatus.code().matches("^[45].+$")) {
             this.report(rqline, "error");
+            if (rsstatus.equals(RsStatus.UNAUTHORIZED)) {
+                this.reportUnauthorized(rqline, rqheaders);
+            }
         } else {
             this.report(rqline, "success");
         }
     }
 
+    /**
+     * Report unauthorized response to metrics.
+     *
+     * @param rqline Request line.
+     * @param rqheaders Request headers.
+     */
+    private void reportUnauthorized(
+        final String rqline,
+        final Iterable<Map.Entry<String, String>> rqheaders
+    ) {
+        if (new RqHeaders(rqheaders, Authorization.NAME).isEmpty()) {
```

Comment:
@olegmoz how about creating `RqHeaders.contains(header)` ?

Reply:
@Vatavuk do not think that it simplifies code significantly. Plus we usually need to read header value, not just see if it is there.

Reply:
@olegmoz the if statement is not readable, I still believe it would be better to have this l inside RqHeaders, but you can instead extract it to a private method

Reply:
@Vatavuk I agree that it is not ideal, but extracting a method for single line feels like an overkill